tests/schemer/sample.py

```diff
@@ -111,3 +111,28 @@ def valid_doc(overrides=None):
         doc.update(overrides)
     return doc
 
+
+def example_validator_1():
+    def validate(value):
+        import logging
+        num_keys = len(value.keys())
+        if not num_keys > 2:
+            return 'At least 2 keys must be provided, only {} were'.format(num_keys)
+    return validate
+
+def example_validator_2():
+    def validate(value):
+        num_keys = len(value.keys())
+        if not num_keys < 4:
+            return 'At most 3 keys must be provided, {} were provided'.format(num_keys)
+    return validate
+
+validator_schema_1 = Schema({'has_json': {'type': bool},
+                             'has_xml': {'type': bool},
+                             'has_yml': {'type': bool}},
+                             validators=example_validator_1())
+validator_schema_2 = Schema({'has_json': {'type': bool},
+                             'has_xml': {'type': bool},
+                             'has_yml': {'type': bool},
+                             'has_other': {'type': bool}},
+                             validators=[example_validator_1(), example_validator_2()])
```

Comment:
Where possible, I've tried to keep examples of the various features we're testing contained in the `BlogPost` schema and associated valid doc. Couple of reasons for this:

1. The sample is to some extent supposed to serve as a realistic example which illustrates how all the various features would be used in a real use case (examples are often better than explanations at communicating ideas). 
1. It's essentially an integration test which ensures that each feature works not just in isolation but also when used together with other features. 

Could we work this into the main example. Maybe add a `modification_date` field and add a doc-level validator which ensures that it's always `>= creation_date`... or something.